### PR TITLE
[wip] Basic Cart configuration support

### PR DESCRIPTION
**Pull request by willock-stripe**
Hi folks! New Stripe.

I grabbed a couple of Terminal devices on my way out of San Francisco and wanted to have a play around with them to build some familiarity. Kudos on the getting started documentation and the helpful example backend – that got me going in no time.

I thought it might be useful to be able to modify the reader display configuration dynamically, and pass that onto the backend. If nothing else, it would make it possible to test the various decline codes using the demonstration app.

Stripe Terminal demo with cart configuration

IMG_3374

### Changes
- Added `CartForm` component which handles cart display config. Moved the `updateLineItems` workflow to the `CartForm` component.
- Moved `workFlow` functionality to parent component so it can be better shared between `CartForm` and `CommonWorkflows` components. Definitely open to how this can be handled better!
- Added `Select` component for currencies.
- `SetReaderDisplay` and `PaymentIntent` methods dynamically hydrated.

### CODE REVIEW

File: src/Forms/CartForm.jsx [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+import Section from "../components/Section/Section.jsx";
+import Text from "../components/Text/Text.jsx";
+import TextInput from "../components/TextInput/TextInput.jsx";
+import Select from "../components/Select/Select.jsx";
+
+class CartForm extends React.Component {
+  static CURRENCIES = [
+    { value: "gbp", label: "GBP" },
+    { value: "usd", label: "USD" }
+  ];
```
Comment (willock-stripe) [inline src/Forms/CartForm.jsx]: I'm not entirely certain what to do with this – should it be driven by the account's supported currencies from the example backend, or just a static list like this?

File: src/MainPage.jsx [hunk 2]
```diff
@@ -25,10 +24,32 @@ class App extends Component {
       reader: null,
       readerLabel: "",
       registrationCode: "",
-      cancelablePayment: false
+      cancelablePayment: false,
+      chargeAmount: 5100,
+      itemDescription: "Red t-shirt",
```
Comment (willock-stripe) [inline src/MainPage.jsx]: An obvious future refactor would be to introduce collections of items to help test the reader UI more thoroughly.

File: src/MainPage.jsx [hunk 3]
```diff
@@ -309,13 +337,35 @@ class App extends Component {
...
+        <>
+          <CommonWorkflows
+            workFlowDisabled={this.isWorkflowDisabled()}
+            onClickCollectCardPayments={() =>
+              this.runWorkflow("collectPayment", this.collectCardPayment)
+            }
+            onClickSaveCardForFutureUse={() =>
+              this.runWorkflow("saveCard", this.saveCardForFutureUse)
+            }
+            onClickCancelPayment={this.cancelPendingPayment}
```
Comment (willock-stripe) [inline src/MainPage.jsx]: Is there a reason that cancelling a pending payment is not a "workflow" in the same sense that collecting payment is?

File: src/Forms/CartForm.jsx (hunk 1 above)
Comment (bg-stripe) [reply src/Forms/CartForm.jsx]: I think a static list is fine for now!

File: src/MainPage.jsx (hunk 2 above)
Comment (bg-stripe) [reply src/MainPage.jsx]: 👍

File: src/MainPage.jsx (hunk 3 above)
Comment (bg-stripe) [reply src/MainPage.jsx]: I think it's because this is a smaller action, and tied to the "collect payment" workflow.

**Pull request merged**